primitives/staking/src/lib.rs

```diff
@@ -37,6 +37,22 @@ pub type SessionIndex = u32;
 /// Counter for the number of eras that have passed.
 pub type EraIndex = u32;
 
+/// Representation of a staking account, which may be a stash or controller account.
+///
+/// Note: once the controller is completely deprecated, this enum can also be deprecated in favor of
+/// the stash account.
+#[derive(Copy, Clone, Debug)]
+pub enum StakingAccount<AccountId> {
+	Stash(AccountId),
+	Controller(AccountId),
+}
+
+impl<AccountId> From<AccountId> for StakingAccount<AccountId> {
+	fn from(account: AccountId) -> Self {
+		StakingAccount::Stash(account)
```

Comment:
Hmm, okay, but could be error prone as well. Why is stash the default? 

Some interfaces use stash, some controller, why favor one over the other?

Reply:
Perhaps because Stash is going away?

Reply:
> Perhaps because Controller is going away?
When the controller is completely deprecated, this whole type will be removed since we don't need to distinguish between stash and controller anymore.

This default is mostly used in test., to avoid verbosity. To make this more future proof, I will remove the default impl and feature gate it for tests only.

Reply:
I gated the `From` implementation with `#[cfg(not(feature = "no-std"))]`, assuming that it will be compiled for benchmarks and tests/mock tests only.